src/stores/app-version-api.ts

```diff
@@ -44,58 +44,36 @@ type GetLastSeenVersionQueryResult = TypedUseQueryStateResult<
 	BaseQueryFn
 >;
 
+function isGreaterExceptPatch( versionA: string | undefined, versionB: string ): boolean {
+	if ( ! versionA ) {
+		return true;
+	}
+
+	if (
+		! semver.valid( versionA ) ||
+		! semver.valid( versionB ) ||
+		semver.gte( versionA, versionB )
+	) {
+		return false;
+	}
+
+	const a = semver.parse( versionA )!;
+	const b = semver.parse( versionB )!;
+
+	if ( a.major === b.major && a.minor === b.minor && a.patch !== b.patch ) {
```

Comment:
For significant prereleases, they will share the same patch and therefore skip this if.